TMDB Movies/Controllers/LoadingViewController.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  LoadingViewController.swift
+//  TMDB Movies
+//
+//  Created by Augusto Ramalho Ramos on 20/02/18.
+//  Copyright © 2018 AugustoRamalhoRamos. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class LoadingViewController: UIViewController {
+    private lazy var activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .whiteLarge)
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        activityIndicator.color = .gray
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(activityIndicator)
+        
+        NSLayoutConstraint.activate([
+            activityIndicator.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            activityIndicator.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+            ])
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
```

Comment:
In most cases yes, especially in cases of async executions, because I don't know if when the execution finishes if I still have access to self . But in case of a view animation there is no need for that.